chore: remove legacy packages and update website layout

GermainBergeron opened the pull request:
### Proposed Changes

Removed legacy packages and updated documentation website layout

### Potential Breaking Changes

Removed plasma-style and plasma-react packages. They are still in the v53 branch

### Acceptance Criteria

-   [ ] The proposed changes are covered by unit tests
-   [ ] The potential breaking changes are clearly identified
-   [ ] [README.md](https://github.com/coveo/plasma/blob/master/README.md) is adjusted to reflect the proposed changes (if relevant)

FelixBlaisThon commented:
The `Deprecated component` section in plasma-website is not working anymore. (Did we want to keep that section with the official deprecation of those packages?) 
<img width="1716" alt="image" src="https://github.com/coveo/plasma/assets/52010042/d6f3e62f-530f-41fd-a82e-da3756b7b581">

GermainBergeron commented:
I forgot to remove the link at the bottom of the side nav. The goal is to remove everything to those legacy packages

gdostie reviewed (commented):
The color page is having a stroke. It can be fixed in a separate PR, just wanted to mention it.

![image](https://github.com/coveo/plasma/assets/35579930/694fb6bd-64eb-4d86-95d4-fdc3af23eea5)

GermainBergeron commented:
@FelixBlaisThon I removed the Deprecated components link from the side nav
@gdostie I fixed the Colors page, it's not perfect but at least we can see the values
